components/core/src/package/install.rs

```diff
@@ -236,7 +236,14 @@ impl PackageInstall {
 
         // Only insert a PATH entry if the resulting path string is non-empty
         if !rooted_path.is_empty() {
-            env.insert(PATH_KEY.to_string(), rooted_path);
+            if cfg!(windows) {
```

Comment:
Again, some documentation about why this is important would be very helpful, particularly given all the subtle discussion in #6657. It can be incredibly difficult to understand the reasons for some of these changes when coming to the code a year later if those reasons aren't clearly documented.